convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,133 @@
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("CleanupModulesOnExit option enabled in %s" % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # Default to return False as it is possible that the CleanupModulesOnExit
+    # is set to no in the config already.
+    return False
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id != "oracle" and system_info.version.major != 8 and system_info.version.minor < 8:
+            logger.info("Skipping the check as it is relevant only for Oracle Linux 8.8 and above.")
+            return
+
+        if not systeminfo.is_systemd_managed_service_running("firewalld"):
+            logger.info("Firewalld service reported that it is not running.")
+            return
+
+        if _is_modules_cleanup_enabled():
+            self.set_result(
+                level="ERROR",
+                id="FIREWALLD_MODULES_CLEANUP_ON_EXIT_CONFIG",
+                title="Firewalld is set to cleanup modules after exit.",
+                description="Firewalld running on Oracle Linux 8 can lead to a conversion failure.",
+                diagnosis=(
+                    "We've detected that firewalld unit is running and that causes iptables and nftables "
+                    "failures on Oracle Linux 8 and under certain conditions it can lead to a conversion failure."
+                ),
+                remediation=(
+                    "Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf "
+                    "to no prior to running convert2rhel:\n"
+                    " sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf\n && firewall-cmd --reload"
+                    "You can change the option back to yes after the system reboot "
+                    "- that is after the system boots into the RHEL kernel."
+                ),
+            )
+        else:
+            # Firewalld is running but the configuration for CleanupModulesOnExit was not set to true/yes
+            description = (
+                "We've detected that firewalld is running and we couldn't find "
+                "check for the CleanupModulesOnExit configuration. "
+                "This means that a reboot will be necessary after the conversion is done to reload the "
+                "kernel modules and prevent firewalld from stop working."
+            )
+            logger.warning(description)
+            self.add_message(
+                level="WARNING",
+                id="FIREWALLD_IS_RUNNING",
+                title="Firewalld is running",
+                description=description,
+            )
+            return
```

Comment:
```suggestion
```
I don't think there's anything to warn about. This `else:` branch is just for the case when the CleanupModulesOnExit is explicitely set to no. And in that case there should be no problem with the firewalld during or after the conversion, if I understood it correctly.
Does it match your testing, @kokesak, @danmyway?

Reply:
@bocekm that's correct, when the `firewalld` is running and the `CleanupModulesOnExit=no` then everything is fine.

Reply:
also the message `we couldn't find check for the CleanupModulesOnExit configuration.` doesn't make sense if the option is present and set to `no`